design/design-generic-references.md

```diff
@@ -0,0 +1,319 @@
+# Generic Resource References
+
+* Owner: Predrag Knezevic (@pedjak)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+Crossplane providers provision and monitor external resources in third party
+cloud providers, following the declared specification in the corresponding managed
+resources. Very often, creation of these objects cannot happen in parallel, i.e. some
+fields can be set only after a given dependency is created.
+
+[Cross resource referencing] solves this problem for majority of cases by
+following [Kubernetes API object reference] convention:
+
+```yaml
+spec:
+  fooRef:
+    name: bar
+```
+
+Field `spec.fooRef` contains the name of the resource where the value for `spec.foo` is to
+be found. The kind of referred resource and the field containing the value used later for
+setting `spec.foo` is determined during compile time.
```

Comment:
Had to read a lot of what "compile time" means here. From https://github.com/crossplane/crossplane/blob/master/design/one-pager-cross-resource-referencing.md I figure this rather means "specified in provider code".

Reply:
done.